lib/samples/apply_style_to_wms_layer/apply_style_to_wms_layer.dart

```diff
@@ -0,0 +1,164 @@
+                  // Add a map view to the widget tree and set a controller.
+                  child: ArcGISMapView(
+                    controllerProvider: () => _mapViewController,
+                    onMapViewReady: onMapViewReady,
+                  ),
+                ),
+                // Build the bottom menu.
+                buildBottomMenu(),
+              ],
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            Visibility(
+              visible: !_ready,
+              child: const SizedBox.expand(
+                child: ColoredBox(
+                  color: Colors.white30,
+                  child: Center(child: CircularProgressIndicator()),
+                ),
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget buildBottomMenu() {
+    return Center(
+      // A drop down button for selecting style.
+      child: DropdownButton(
+        alignment: Alignment.center,
+        hint: const Text(
+          'Choose a style',
+          style: TextStyle(
+            color: Colors.deepPurple,
+          ),
+        ),
+        icon: const Icon(Icons.arrow_drop_down),
+        iconEnabledColor: Colors.deepPurple,
+        iconDisabledColor: Colors.grey,
+        style: const TextStyle(color: Colors.deepPurple),
+        value: _selectedStyle,
+        items: _stylesTitles.map((items) {
+          return DropdownMenuItem(
+            value: items,
+            child: Text(items),
+          );
+        }).toList(),
+        onChanged: (style) {
+          if (style != null) {
+            changeStyle(style);
+          }
+        },
+      ),
+    );
+  }
+
+  void changeStyle(String style) async {
+    // Set the selected style.
+    setState(() => _selectedStyle = style);
+
+    // Get the available styles from the first sublayer.
+    final firstSublayer = _wmsLayer.sublayers.first as WmsSublayer;
+    final styles = firstSublayer.sublayerInfo.styles;
+
+    switch (style) {
+      case 'Default':
+        // Apply the first style to the first sublayer.
+        setState(() => firstSublayer.currentStyle = styles[0]);
+      case 'Contrast stretch':
+        // Apply the second style to the first sublayer.
+        setState(() => firstSublayer.currentStyle = styles[1]);
+      default:
+        throw StateError('Unknown style');
+    }
+  }
+
+  void onMapViewReady() async {
+    // Create a map with spatial reference appropriate for the service.
+    final map = ArcGISMap(spatialReference: SpatialReference(wkid: 26915));
+    map.minScale = 7000000.0;
+    // Create a new WMS layer displaying the specified layers from the service.
+    // The default styles are chosen by default.
+    _wmsLayer = WmsLayer.withUriAndLayerNames(
+      uri: Uri.parse(
+        'https://imageserver.gisdata.mn.gov/cgi-bin/mncomp?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetCapabilities',
+      ),
+      layerNames: ['mncomp'],
+    );
```

Comment:
```suggestion
   const wmsLayerUri = 'https://imageserver.gisdata.mn.gov/cgi-bin/mncomp?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetCapabilities';
    _wmsLayer = WmsLayer.withUriAndLayerNames(
      uri: Uri.parse(wmsLayerUri),
      layerNames: ['mncomp'],
    );
```
This is a nit-pick and not incorrect, but follows a pattern we've been using in other samples.